srml/contracts/src/wasm/prepare.rs

```diff
@@ -74,6 +74,23 @@ impl<'a> ContractModule<'a> {
 		Ok(())
 	}
 
+	/// Ensures that tables declared in the module are not too big.
+	fn ensure_table_size_limit(&self, limit: u32) -> Result<(), &'static str> {
+		let err_msg = self.module.table_section()
+			.and_then(|table_section| table_section.entries().first())
+			.and_then(|table_type| {
+				if table_type.limits().initial() > limit {
```

Comment:
I wonder why we use `initial` here, wouldn't it be more appropriate using `maximum`?

Reply:
`initial` is the actual size of the table since there is no way to grow the table.

Per https://webassembly.org/docs/modules/#table-section,

> In the MVP, tables can only be resized via host-defined APIs (such as the JavaScript WebAssembly.Table.prototype.grow). A grow_table may be added in the future :unicorn:. In either case, table growth is guaranteed to fail if attempting to grow past the declared maximum. As with linear memory, when a maximum is declared, implementations should (non-normative) attempt to reserve virtual memory up to the maximum size. While failure to allocate the initial memory size is a runtime error, failure to reserve up to the maximum is not. When a maximum memory size is not declared, on architectures with limited virtual address space, engines should allocate only the initial size and reallocate on demand.

To avoid confusion, it may be worth further enforcing that maximum is either omitted or equal to initial.

Reply:
Yeah, I just want to open up a discussion. Note that we use `maximum` for memories and reject the binaries that don't have it (and thus, the modules need some postprocessing before they could be deployed on a chain. At the moment it is done [really messy](https://github.com/paritytech/ink/blob/9589899b5630c42870de24b0fdb9e67587531f70/cli/template/build.sh#L12))